src/types.ts

```diff
@@ -1,12 +1,18 @@
-export type PersonSendMessage = {
+export type TextMessage = {
     message: string,
     userName: string,
     clientId: string,
+    createdAt: string,
 }
 
-export type PersonSendImage = {
-    src: {
-        result: string,
-        clientId: string
-    }
-}
\ No newline at end of file
+export type ImageMessage = {
+    result: string,
+    clientId: string,
+    createdAt?: never,
+    userName?: never,
+    message?: never
```

Comment:
why never? you should use Interface: https://www.typescriptlang.org/docs/handbook/2/objects.html|
find common props and extend it in `ImageMessage` and `TextMessage`